OSVR-Unity/Assets/OSVRUnity/src/DisplayController.cs

```diff
@@ -0,0 +1,248 @@
+﻿/// OSVR-Unity Connection
+///
+/// http://sensics.com/osvr
+///
+/// <copyright>
+/// Copyright 2015 Sensics, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+/// </copyright>
+/// <summary>
+/// Author: Greg Aring
+/// Email: greg@sensics.com
+/// </summary>
+using UnityEngine;
+using System.Collections;
+
+namespace OSVR
+{
+    namespace Unity
+    {
+        //This class is responsible for creating the head, eyes, and surfaces in our scene.
+        //Rendering parameters are obtained from ClientKit.
+        //DisplayController creates VRViewer and VREyes as children. Each eye has a VRSurface child with a camera.
+        //In this implementation, we are assuming that there is exactly one viewer and one surface per eye.
+        public class DisplayController : MonoBehaviour
+        {
+            public const uint DEFAULT_VIEWER = 0; //assume exactly one viewer in this Unity implementation
+            public const uint DEFAULT_SURFACE = 0; //assume exactly one viewer in this Unity implementation
+            public const uint NUM_VIEWERS = 1;
+            public const uint NUM_SURFACE_PER_EYE = 1; 
+
+            private ClientKit _clientKit;
+            private OSVR.ClientKit.DisplayConfig _displayConfig;
+            private VRViewer[] viewers;
+            private VREye[] eyes;
+            private uint _eyeCount;
+            private uint _viewerCount;
+            private bool renderedStereo = false;
+            private bool displayConfigInitialized = false;
+
+            public OSVR.ClientKit.DisplayConfig DisplayConfig
+            {
+                get { return _displayConfig; }
+                set { _displayConfig = value; }
+            }          
+            public VRViewer[] Viewers { get { return viewers; } }           
+            public VREye[] Eyes { get { return eyes; } }
+            public uint EyeCount { get { return _eyeCount; } }
+            public uint ViewerCount { get { return _viewerCount; } }
+            public float nearClippingPlane = 0.01f;
+            public float farClippingPlane = 1000f;
+
```

Comment:
Asked about IPD because Cardboard allows FOV to be changed for special effect: https://github.com/googlesamples/cardboard-unity/blob/91bdb1c00ac35e14f0f403bc4ede49670dad8008/Cardboard/Scripts/StereoController.cs#L76

Could be used in a game if the player could change size (would a smaller IPD make everything look bigger?). Agreed that it shouldn't really be tampered with in most cases.